app/uk/gov/hmrc/economiccrimelevyreturns/controllers/ReturnSubmittedController.scala

```diff
@@ -45,11 +45,11 @@ class ReturnSubmittedController @Inject() (
     with I18nSupport
     with ErrorHandler {
 
-  def onPageLoad: Action[AnyContent] = (authorise andThen getReturnData).async { implicit request =>
+  def onPageLoad: Action[AnyContent] = authorise.async { implicit request =>
     val chargeReference: Option[String] = request.session.get(SessionKeys.ChargeReference)
     val amountDue: BigDecimal           = BigDecimal(request.session(SessionKeys.AmountDue))
-    val obligationDetails               = request.eclReturn.obligationDetails
-    val email                           = request.eclReturn.contactEmailAddress
+    val obligationDetails               = ObligationDetails.read(request.session.get(SessionKeys.ObligationDetails))
```

Comment:
Do we want to use the valueOrError on here, like registration has?

Reply:
Resolved, all of these fields are handled in for comprehension below.